File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if(root == nullptr)
            return {};
        return solve(root);
    }
private:
    vector<int> solve(TreeNode* root) {
        vector<int> ans;
        queue<TreeNode*> queue;
        queue.push(root);
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            
            for(int i = 0; i < levelSize; ++i) {
                auto curr = queue.front();
                queue.pop();
                if(curr -> left) queue.push(curr -> left);
                if(curr -> right) queue.push(curr -> right);
                if(i == levelSize - 1) ans.push_back(curr -> val);
            }
        }
        return ans;
    }
};

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        ans = []

        queue = deque()
        queue.append(root)

        while len(queue) > 0:
            levelSize = len(queue)

            for i in range(levelSize):
                curr = queue.popleft()
                if curr.left: queue.append(curr.left)
                if curr.right: queue.append(curr.right)

                if i == levelSize - 1:
                    ans.append(curr.val)
        return ans
        